go/libraries/doltcore/env/environment.go

```diff
@@ -1115,8 +1129,12 @@ func (dEnv *DoltEnv) GetUserHomeDir() (string, error) {
 	return getHomeDir(dEnv.hdp)
 }
 
-func (dEnv *DoltEnv) TempTableFilesDir() string {
-	return mustAbs(dEnv, dEnv.GetDoltDir(), tempTablesDir)
+func (dEnv *DoltEnv) TempTableFilesDir() (string, error) {
+	doltDir := dEnv.GetDoltDir()
+	if doltDir == "" {
+		return "", ErrDoltRepositoryNotFound
+	}
+	return mustAbs(dEnv, doltDir, tempTablesDir), nil
```

Comment:
it might make sense to just do the extra error check here, now that we're returning an error

Reply:
I think it would be slower to check if the output of `mustAbs` method has substring of `.dolt` than to check if `doltDir == ""`

Reply:
i mean you can just put `return absPath, err := dEnv.FS.Abs(filepath.Join(path...))` there? no real reason to panic

Reply:
Ah I see, makes sense.